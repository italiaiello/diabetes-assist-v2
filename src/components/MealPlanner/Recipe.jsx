import React from 'react'
import { useRecipeFetch } from '../../hooks/DisplayMealRecipe'

const Recipe = props => {

    console.log(props.slicedRecipeData[props.counter])

    const id = props.slicedRecipeData[props.counter].id

    const [isLoading, recipeData] = useRecipeFetch(`https://api.spoonacular.com/recipes/${id}/analyzedInstructions?apiKey=d464b770a6c2452cb7d56fc6ccea9eb5`, id)

    console.log(recipeData)

    return (
        isLoading || !recipeData.length ?
        <h1>Fetching Recipe...</h1>
        :
        
        <article className="pageDisplay">
            <div className="backAndHeading">
                <div className="sectionHeading">
                    <div className="back">
                        <div className="leftArrow" onClick={() => props.onRouteChange('meal')}></div>
                        <p>Back</p>
                    </div>
                    <h1>{props.slicedRecipeData[props.counter].title}</h1>
                </div>
            </div>
            <article id="recipeSection">
                <article id="mealImage">
                    <figure>
                        <img src={`https://spoonacular.com/recipeImages/${props.slicedRecipeData[props.counter].image}`} alt={props.slicedRecipeData[props.counter].title} />
                    </figure>
                    <button id="sendButton">Send ingredients list and recipe to your phone</button>
                </article>
                <article id="recipeInfoContainer">
                    <h3>Recipe</h3>
                    <article className="recipeInfo">
                        
                        {
                            recipeData[0].steps.map((step, i) => {
                                return (
                                    <article key={i} className="step">
                                        <p className="stepNumber">{`${step.number}.`}</p>
                                        <p className="stepDesc">{step.step}</p>
                                    </article>
                                    
                                )
                            })
                        }
                    </article>
                </article>
            </article>
        </article>
    )
}

export default Recipe