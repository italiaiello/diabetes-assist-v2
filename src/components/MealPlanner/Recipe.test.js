import React from "react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Recipe from './Recipe';

let route = ""

const onRouteChange = (newRoute) => {
  route = newRoute;
}

const mockSlicedRecipeData = [
    {
        "id": 592479,
        "title": "Kale and Quinoa Salad with Black Beans",
        "readyInMinutes": 50,
        "servings": 6,
        "image": "Kale-and-Quinoa-Salad-with-Black-Beans-592479.jpg",
        "imageUrls": [
            "Kale-and-Quinoa-Salad-with-Black-Beans-592479.jpg"
        ]
    },
    {
        "id": 547775,
        "title": "Creamy Avocado Pasta",
        "readyInMinutes": 15,
        "servings": 2,
        "image": "Creamy-Avocado-Pasta-547775.jpg",
        "imageUrls": [
            "Creamy-Avocado-Pasta-547775.jpg"
        ]
    },
    {
        "id": 818941,
        "title": "Avocado Toast with Eggs, Spinach, and Tomatoes",
        "readyInMinutes": 10,
        "servings": 1,
        "image": "avocado-toast-with-eggs-spinach-and-tomatoes-818941.jpg",
        "imageUrls": [
            "avocado-toast-with-eggs-spinach-and-tomatoes-818941.jpg"
        ]
    },
]

const mockCounter = 1

const mockProps = {
    onRouteChange: onRouteChange,
    slicedRecipeData: mockSlicedRecipeData,
    counter: mockCounter
}

describe("Recipe.js", () => {
    let wrapper;

    it('renders', () => {
        wrapper = shallow(<Recipe {...mockProps} />);
        expect(toJson(wrapper)).toMatchSnapshot()
    })

});