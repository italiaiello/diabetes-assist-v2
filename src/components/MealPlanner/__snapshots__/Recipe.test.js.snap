// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe.js renders 1`] = `
<h1>
  Fetching Recipe...
</h1>
`;
